utils/update_copyright.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+# encoding: utf-8
+
+###
+# Update the copyright notices year in all C++ files in src/
+###
+
+from datetime import date
+import os
+import re
+import sys
+
+previous_year = str(date.today().year-1)
+if len(sys.argv) == 2:
+    previous_year = sys.argv[1]
+
+new_year = str(date.today().year)
+
+previous_notice = "^\tCopyright \(C\) 20[0-9]{2} - "+previous_year+"$"
```

Comment:
Is this really right? It only matches start years in the range [2000,2100), right?

Also as a side note, I generally recommend using raw string syntax for regular expressions.

Reply:
Wesnoth was initially created in 2003, so it wouldn't make much sense for there to be copyright notices for earlier years.

Reply:
Sure, but not supporting 2100 and above is also kind of silly when it's trivial to support it. I have no idea if Wesnoth will still be around in 2100, but all you have to do is change this to `[0-9]{4}` and it automatically works if it still is.

Reply:
Mainly it's to catch some cases of typoes, like if someone typed 2201 instead of 2021.

Reply:
Also for some reason
`r"^	Copyright (C) 20[0-9]{2} - "+previous_year+r"$"`
does not work while
`"^\tCopyright \(C\) 20[0-9]{2} - "+previous_year+"$"`
does work.

Reply:
Well, space and tab are different characters. If you want to match any whitespace you could use `\s`. `()` denotes a capture group but you want to match literal parentheses so you need to escape.

Reply:
If I copy/paste that character into a text editor it gets treated as a tab.  No idea why it only shows up here as taking up a single space.